File: apps/backend/src/db/tool_event_store_test.go
package db

import (
	"testing"
)

func TestToolEventStore_RecordAndList(t *testing.T) {
	database := testDB(t)
	store := NewToolEventStore(database)

	err := store.Record("session-1", "Edit", "/src/main.go")
	if err != nil {
		t.Fatalf("Record: %v", err)
	}
	err = store.Record("session-1", "Bash", "")
	if err != nil {
		t.Fatalf("Record: %v", err)
	}

	events, err := store.ListBySession("session-1", 50)
	if err != nil {
		t.Fatalf("ListBySession: %v", err)
	}
	if len(events) != 2 {
		t.Errorf("got %d events, want 2", len(events))
	}
	if events[0].ToolName != "Edit" {
		t.Errorf("ToolName = %q, want %q", events[0].ToolName, "Edit")
	}
}

File: apps/orchestrator/src/db/stores/__tests__/tool-event-store.test.ts
import { describe, test, expect, beforeEach } from "bun:test";
import type { Database } from "bun:sqlite";
import { openMemory } from "../../database.ts";
import { ToolEventStore } from "../tool-event-store.ts";

describe("ToolEventStore", () => {
  let db: Database;
  let store: ToolEventStore;

  beforeEach(() => {
    db = openMemory();
    store = new ToolEventStore(db);
  });

  test("Create inserts and returns a tool event", () => {
    const e = store.Create("session-1", "Read", "/src/main.ts");
    expect(e.id).toBeGreaterThan(0);
    expect(e.session_id).toBe("session-1");
    expect(e.tool_name).toBe("Read");
    expect(e.file_path).toBe("/src/main.ts");
    expect(e.timestamp).toBeTruthy();
  });

  test("ListBySession returns events ordered by timestamp", () => {
    store.Create("session-1", "Read", "/a.ts");
    store.Create("session-1", "Edit", "/b.ts");
    store.Create("session-2", "Grep", "/c.ts");

    const events = store.ListBySession("session-1");
    expect(events).toHaveLength(2);
    expect(events[0].tool_name).toBe("Read");
    expect(events[1].tool_name).toBe("Edit");
  });

  test("ListBySession returns empty for unknown session", () => {
    expect(store.ListBySession("nonexistent")).toHaveLength(0);
  });
});
